src/services/http/controllers/Github.ts

```diff
@@ -0,0 +1,88 @@
+import { NextFunction, Request, RequestHandler, Response, Router } from 'express';
+import passport from 'passport';
+import { Strategy as GhStrategy } from 'passport-github2';
+import { calculateReputation, OAuthProvider } from '@interep/reputation';
+
+import { Controller } from './interface';
+import config from '../../../util/config';
+import { getReceivedStars } from '../../../util/github';
+import { makeResponse } from '../utils';
+
+type GhUser = {
+  username: string;
+  _json: {
+    followers: number;
+    plan: { name: string };
+  };
+};
+
+export class GithubController extends Controller {
+  redirectUrl: string;
+
+  constructor() {
+    super();
+    passport.use(
+      new GhStrategy(
+        {
+          clientID: config.ghClientId,
+          clientSecret: config.ghClientSecret,
+          callbackURL: config.ghCallbackUrl,
+        },
+        async (accessToken: string, refreshToken: string, profile: GhUser, done: any) => {
+          const {
+            username,
+            _json: {
+              followers,
+              plan: { name: planName },
+            },
+          } = profile;
+
+          const proPlan = planName === 'pro';
+          const receivedStars = await getReceivedStars(username);
+          const reputation = calculateReputation(OAuthProvider.GITHUB, {
+            followers,
+            receivedStars,
+            proPlan,
+          });
+
+          const githubAuthDB = await this.call('db', 'getGithubAuth');
```

Comment:
should we `try...catch` and call `done(error, null)` here? otherwise, not sure if this would fail silently or with logs

all else looks good!

Reply:
Right, I think it happened to me.
It should happen only while developing though imo, when the auth table may not be there yet...

Reply:
https://github.com/zkitter/zkitterd/pull/54/commits/af56d4510781fdcbb5be3ad19285ced0c746cc75